src/pages/announcements/utils.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import yup from '../../utils/yup-config';
+import { Duration, parse, serialize } from 'tinyduration';
+
+export const ID = 'id';
+export const DATE = 'creationDate';
+export const MESSAGE = 'message';
+export const DURATION = 'duration';
+export const DAYS = 'days';
+export const HOURS = 'hours';
+export const MINUTES = 'minutes';
+
+export interface DurationFormData {
+    [DAYS]: number | null;
+    [HOURS]: number | null;
+    [MINUTES]: number | null;
+}
+
+export interface AnnouncementFormData {
+    [MESSAGE]: string;
+    [DURATION]: DurationFormData;
+}
+
+export const emptyFormData: AnnouncementFormData = {
+    [MESSAGE]: '',
+    [DURATION]: {
+        [DAYS]: null,
+        [HOURS]: null,
+        [MINUTES]: null,
+    },
+};
+
+// const formSchema: ObjectSchema<FormData> = yup
+export const formSchema = yup
```

Comment:
Can we separate the formSchema from type definitions ?

Reply:
What do you mean ?